scripts/dotnet-cli-build/CompileTargets.cs

```diff
@@ -85,45 +84,78 @@ public static BuildTargetResult Compile(BuildTargetContext c)
         [Target]
         public static BuildTargetResult GenerateStubHostPackages(BuildTargetContext c)
         {
-            string currentRid = GetRuntimeId();
             var buildVersion = c.BuildContext.Get<BuildVersion>("BuildVersion");
-            PrepareDummyRuntimeNuGetPackage(
-                            DotNetCli.Stage0,
-                            buildVersion.HostNuGetPackageVersion,
-                            Dirs.CorehostDummyPackages);
-            foreach (var hostPackageId in HostPackages)
+            string currentRid = GetRuntimeId();
+            PrepareDummyRuntimeNuGetPackage(DotNetCli.Stage0);
+            foreach (var hostPackage in buildVersion.LatestHostPackages)
             {
-                foreach (var rid in HostPackageSupportedRids)
+                foreach (var rid in HostPackageSupportedRids.Keys)
                 {
                     if (!rid.Equals(currentRid))
                     {
                         CreateDummyRuntimeNuGetPackage(
                             DotNetCli.Stage0,
-                            hostPackageId,
+                            hostPackage.Key,
                             rid,
-                            buildVersion.HostNuGetPackageVersion,
+                            hostPackage.Value,
                             Dirs.CorehostDummyPackages);
                     }
                 }
             }
             return c.Success();
         }
 
-        private static string HostVer = "1.0.1";
-        private static string HostPolicyVer = "1.0.1";
-        private static string HostFxrVer = "1.0.1";
+        [Target(nameof(PrepareTargets.Init))]
+        public static BuildTargetResult RestoreLockedCoreHost(BuildTargetContext c)
```

Comment:
Locking down corehost (i.e., restores and publishes a known version of corehost) into `artifacts/osx.10.10-x64/corehost/locked`